controllers/build_controller.go

```diff
@@ -0,0 +1,315 @@
+			imageList = append(imageList, i.Image)
+			log.Log.Info(fmt.Sprintf("    Found image: %v\n", i.Image))
+		}
+	}
+
+	return imageList, nil
+}
+
+// generateScanPodSpec generates the pod spec for the scanner that will be injected into the namespace
+func generateScanPodSpec(images []string, scanImageName, buildName, namespace, projectName, environmentName, dockerhost string) (*corev1.Pod, error) {
+
+	if len(images) == 0 {
+		return nil, errors.New("No images to scan")
+	}
+
+	insightScanImages := strings.Join(images, ",")
+
+	// Define PodSpec
+
+	podSpec := &corev1.Pod{
+		ObjectMeta: v12.ObjectMeta{
+			Namespace: namespace,
+			Name:      scannerNameFromBuildname(buildName),
+			Labels:    imageScanPodLabels(),
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "lagoon-deployer",
+			Containers: []corev1.Container{
+				{
+					Name:  "scanner",
+					Image: scanImageName,
+					Env: []corev1.EnvVar{
+						{
+							Name:  "INSIGHT_SCAN_IMAGES",
+							Value: insightScanImages,
+						},
+						{
+							Name:  "NAMESPACE",
+							Value: namespace,
+						},
+						{
+							Name:  "PROJECT",
+							Value: projectName,
+						},
+						{
+							Name:  "ENVIRONMENT",
+							Value: environmentName,
+						},
+						{
+							Name:  "DOCKER_HOST",
+							Value: dockerhost,
+						},
+					},
+					ImagePullPolicy: "Always",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "lagoon-internal-registry-secret",
+							MountPath: "/home/.docker/",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			RestartPolicy: "Never",
+			Volumes: []corev1.Volume{ // Here we have to mount the
+				{
+					Name: "lagoon-internal-registry-secret",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: "lagoon-internal-registry-secret",
+							Items: []corev1.KeyToPath{
+								{Key: ".dockerconfigjson", Path: "config.json"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return podSpec, nil
+}
+
+func (r *BuildReconciler) killExistingScans(ctx context.Context, newScannerName string, namespace string) error {
+	// Find pods in namespace with the image scan pod labels
+	podlist := &corev1.PodList{}
+	ls := client.MatchingLabels(imageScanPodLabels())
+	ns := client.InNamespace(namespace)
+	err := r.Client.List(ctx, podlist, ns, ls)
+	if err != nil {
+		log.Log.Info("Issue generating existing scan list")
+		return err
+	}
+	log.Log.Info("Comparing with: " + newScannerName)
+	for _, i := range podlist.Items {
+
+		log.Log.Info("Looking at following image by name: " + i.Name)
+		//if i.Name != newScannerName { // Then we have a rogue pod
+		err = r.Client.Delete(ctx, &i)
+		if err != nil {
+			return err
+		}
+		log.Log.Info(fmt.Sprintf("Successfully deleted old insights-scanner pod: %v", i.Name))
+		//}
+	}
+	return nil
+}
+
+func imageScanPodLabels() map[string]string {
+	return map[string]string{
+		"insights.lagoon.sh/imagescanner": "scanning",
```

Comment:
can this label have any other value?

Reply:
Right now it doesn't have any other values. I've implied it might in that one TODO where I mention that it's currently fire-and-forget. So presumably we could have it have a "compeleted" status, or particular failed status if we needed.

Reply:
would it make sense to be `insights.lagoon.sh/scan-status` then?

Reply:
yeah, that makes sense. I'll make the change.